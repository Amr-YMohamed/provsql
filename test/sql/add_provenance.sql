\set ECHO none
SET search_path TO public, provsql;

CREATE TYPE classification_level AS ENUM ('unclassified','restricted','confidential','secret','top_secret');

CREATE TABLE personal(
  id SERIAL PRIMARY KEY,
  name varchar,
  position varchar,
  city varchar,
  classification classification_level
);

INSERT INTO personal (name,position,city,classification) VALUES
  ('John','Director','New York','unclassified'),
  ('Paul','Janitor','New York','restricted'),
  ('Dave','Analyst','Paris','confidential'),
  ('Ellen','Field agent','Berlin','secret'),
  ('Magdalen','Double agent','Paris','top_secret'),
  ('Nancy','HR','Paris','restricted'),
  ('Susan','Analyst','Berlin','secret');

SELECT add_provenance('personal');

SELECT attname
FROM pg_attribute
WHERE attrelid ='personal'::regclass AND attnum>1
ORDER BY attname;

