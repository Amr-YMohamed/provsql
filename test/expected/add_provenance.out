\set ECHO none
 add_provenance 
----------------
 
(1 row)

                                    Table "public.personal"
     Column     |         Type         |                       Modifiers                       
----------------+----------------------+-------------------------------------------------------
 id             | integer              | not null default nextval('personal_id_seq'::regclass)
 name           | character varying    | 
 position       | character varying    | 
 city           | character varying    | 
 classification | classification_level | 
 provsql        | provenance_token     | default uuid_generate_v4()
Indexes:
    "personal_pkey" PRIMARY KEY, btree (id)
    "personal_provsql_key" UNIQUE CONSTRAINT, btree (provsql)
Foreign-key constraints:
    "provsqlfk" FOREIGN KEY (provsql) REFERENCES provenance_circuit_gate(gate)
Triggers:
    add_provenance_circuit_gate BEFORE INSERT ON personal FOR EACH ROW EXECUTE PROCEDURE add_provenance_circuit_gate_trigger()

